crates/pet-core/src/lib.rs

```diff
@@ -39,5 +40,5 @@ pub trait Locator: Send + Sync {
     /**
      * Finds all environments specific to this locator.
      */
-    fn find(&self) -> Option<LocatorResult>;
+    fn find(&self, reporter: &dyn Reporter);
```

Comment:
Previously we'd report as soon as each locator finds environments.
Thats very fast.
However I can see conda getting slower and slower.
E.g. everything completes in < 50ms for me, except for conda.
Now that I have installed 30 conda envs, and 8 conda installs, it takes 100-200ms
So, first conda gets reported only after all conda envs have been disccovered.

The change is to now report immediately as they are discovered.
Perf is still the same, perhaps slightly faster,